### PR TITLE
Rebasing dev0.8.1 on main

### PR DESCRIPTION
**Pull request by nfwvogt**

### CODE REVIEW

File: .github/workflows/ci_tests.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ on:
   # push:
   #   branches: [main]
   pull_request:
-    branches: [main]
+    branches: [main, "dev0.8.1"]
```
Comment (kbarkhqs) [inline .github/workflows/ci_tests.yml:7]: Is this safe?

File: .github/workflows/ci_tests.yml [hunk 2]
```diff
@@ -185,46 +181,38 @@ jobs:
...
-          RUSTFLAGS="-C link-arg=-undefined -C link-arg=dynamic_lookup" maturin build -i ${{ matrix.python.interpreter }} --out wheels  -m qoqo-quest/Cargo.toml --locked
-      - name: universal wheels
-        if: ${{ matrix.combinations.runs_on == 'macOS-latest' && matrix.python-version >= '3.8' }}
-        run: |
-          RUSTFLAGS="-C link-arg=-undefined -C link-arg=dynamic_lookup" maturin build -i ${{ matrix.python.interpreter }} --out wheels  -m qoqo-quest/Cargo.toml --locked
-      - name: test install
-        run: |
-          pip install wheels/* --force-reinstall
+  # test_maturin_builds_macos:
+  #   name: maturin_check-${{ matrix.combinations.runs_on }}
```
Comment (kbarkhqs) [inline .github/workflows/ci_tests.yml:185]: Have these been temporarily commented out, or permanently?

File: .github/workflows/python_ci_tests.yml [hunk 3]
```diff
@@ -28,31 +28,23 @@ jobs:
...
-    runs-on: ${{ matrix.combinations.runs_on }}
+    name: maturin_check-ubuntu-${{ matrix.python.py }}
+    runs-on: 'ubuntu-latest'
     strategy:
       matrix:
-        combinations: [
-          { runs_on: 'ubuntu-latest',  rust: "x86_64-unknown-linux-gnu" },
-        ]
         python: [
-            {py: '3.7', interpreter: "python3.7"},
```
Comment (kbarkhqs) [inline .github/workflows/python_ci_tests.yml:44]: Python 3.7 is still officially supported until June, should we keep it until then?

File: qoqo-quest/pyproject.toml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 [project]
 name = "qoqo-quest"
-version = "0.9.0-alpha.2"
+version = "0.9.0"
 dependencies = [
   'numpy',
   'qoqo>=1.2.0a1',
```
Comment (kbarkhqs) [inline qoqo-quest/pyproject.toml:6]: Should this be changed to 1.2.0?

**Review by kbarkhqs — APPROVED** [review]

**Comment by kbarkhqs** [comment]
I believe `cargo licensepull > DEPENDENCIES` hasn't been run yet

File: .github/workflows/ci_tests.yml (hunk 1 above)
Comment (nfwvogt) [reply .github/workflows/ci_tests.yml:7]: Yes, only adds an extra check, we can remove it in the next release

**Pull request merged**